src/test/java/edu/kit/kastel/sdq/artemis4j/grading/penalty/ThresholdPenaltyRuleTest.java

```diff
@@ -0,0 +1,67 @@
+                " 1             | 1           | 0.5     | 2           | -0.5      | true     ",
+                " 1             | 1           | 0.5     | 3           | -0.5      | true     ",
+                " 1             | 1           | 0.5     | 4           | -0.5      | true     ",
+                //
+                " 2             | 1           | 0.5     | 1           | -0.0      | false    ",
+                " 2             | 1           | 0.5     | 2           | -0.5      | false    ",
+                " 2             | 1           | 0.5     | 3           | -0.5      | true     ",
+                " 2             | 1           | 0.5     | 4           | -0.5      | true     ",
+                //
+                " 2             | 3           | 0.5     | 1           | -0.0      | false    ",
+                " 2             | 3           | 0.5     | 2           | -0.5      | false    ",
+                " 2             | 3           | 0.5     | 3           | -0.5      | false    ",
+                " 2             | 3           | 0.5     | 4           | -1.0      | false    ",
+                " 2             | 3           | 0.5     | 5           | -1.0      | false    ",
+                " 2             | 3           | 0.5     | 6           | -1.5      | false    ",
+                " 2             | 3           | 0.5     | 7           | -1.5      | true     ",
+                " 2             | 3           | 0.5     | 8           | -1.5      | true     ",
+                //
+                " 5             | 2           | 0.5     | 1           | -0.0      | false    ",
+                " 5             | 2           | 0.5     | 2           | -0.0      | false    ",
+                " 5             | 2           | 0.5     | 3           | -0.0      | false    ",
+                " 5             | 2           | 0.5     | 4           | -0.0      | false    ",
+                " 5             | 2           | 0.5     | 5           | -0.5      | false    ",
+                " 5             | 2           | 0.5     | 6           | -0.5      | false    ",
+                " 5             | 2           | 0.5     | 7           | -0.5      | false    ",
+                " 5             | 2           | 0.5     | 8           | -0.5      | false    ",
+                " 5             | 2           | 0.5     | 9           | -0.5      | false    ",
+                " 5             | 2           | 0.5     | 10          | -1.0      | false    ",
+                " 5             | 2           | 0.5     | 11          | -1.0      | true     ",
+                " 5             | 2           | 0.5     | 12          | -1.0      | true     ",
+            })
+    void testCalculatePoints(
+            int threshold, Integer repetitions, double penalty, int annotations, double expected, boolean isCapped) {
+        PenaltyRule rule = new ThresholdPenaltyRule(threshold, penalty, repetitions == null ? 1 : repetitions);
```

Comment:
Why do you check against `null` here? How can `null` occur in this parametrized test?

Reply:
One the same note, why does testCalculatePoints recieve repititions as an Integer rather than an int?

Reply:
In these tests you can leave a column empty. This would result in it passing the default value, in the case of an `int` it would be `0` or for an `Integer` it would be `null`. With the current implementation one could leave a column empty, and it would behave like it does when no value is present in the JSON for the `repetitions` key.

Reply:
It is just a test, so I thought the code wouldn't hurt 🤷, but I can remove it if necessary